wagtail/snippets/tests/test_viewset.py

```diff
@@ -164,6 +164,19 @@ def test_deconstruct(self):
         # It should not add any extra kwargs for the icon
         self.assertEqual(kwargs, {"required": False})
 
+    def test_snippet_icon_override_if_custom_icon_is_defined(self):
+        # Snippet icon is overridden if it is defined, FullFeaturedSnippet has a custom icon as "cog"
+        chooser_block = SnippetChooserBlock(FullFeaturedSnippet)
+        self.assertEqual(chooser_block.target_model.snippet_viewset.icon, "cog")
```

Comment:
Issue: Whilst this assertion is useful to confirm the snippet has a custom icon, it doesn't test the block's icon, which is what the issue is resolving.